File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
            unordered_map<int,int> prefixsumcount;
            prefixsumcount[0]=1;
            int prefixsum=0;
            int count=0;
            for(int i=0;i<nums.size();i++)
            {
                prefixsum+=nums[i];
                int remove=prefixsum-k;
                if(prefixsumcount.find(remove)!=prefixsumcount.end())
                {
                    count+=prefixsumcount[remove];
                }
                prefixsumcount[prefixsum]++;
            }
            return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer>mpp=new HashMap<>();
        mpp.put(0,1);
        int prefixsum=0;
        int count=0;
        for(int i=0;i<nums.length;i++)
        {
            prefixsum+=nums[i];
            int remove=prefixsum-k;
            if(mpp.containsKey(remove))
            {
                count+=mpp.get(remove);
            }
            mpp.put(prefixsum,mpp.getOrDefault(prefixsum,0)+1);
        }
        return count;
    }
}